ang/sqlTaskConfigurator.js

```diff
@@ -576,6 +578,351 @@
+          }).then((actions) => {
+            const entityActions = actions.map(({ name }) => ({ name, value: name }));
+            entityActionCache.set(entity, entityActions);
+            return entityActions;
+          }).catch(error => {
+            console.error(error);
+            return [];
+          });
+        }
+
+        function fetchErrorHandlerOptions() {
+          return new Promise((resolve, reject) => {
+            CRM.api3("Sqltaskfield", "get_handle_api_errors_options", {
+              sequential: 1,
+              options: { limit: 0 },
+            }).done(result => {
+              if (result.is_error) {
+                reject(result);
+                return;
+              }
+
+              resolve(
+                Object.entries(result.values[0]).map(
+                  ([ value, name ]) => ({ value, name })
+                )
+              );
+            });
+          }).catch(error => {
+            console.error(error);
+            return [];
+          });
+        }
+
+        async function initialSetup() {
+          await addUrlInputControls();
+
+          let entity = $scope.model.entity;
+
+          $scope.apiv4Entities = await fetchEntities();
+          $scope.apiv4ErrorHandlerOptions = await fetchErrorHandlerOptions();
+          $scope.$apply();
+
+          if (!isValidEntity(entity)) {
+            entity = undefined;
+          }
+
+          await setApiUrl("");
+          await selectEntity(entity);
+          await updateAction(entity);
+        }
+
+        function isValidEntity(entity) {
+          return $scope.apiv4Entities.find(({ name }) => name === entity) !== undefined;
+        }
+
+        function parseApiExplorerUrl(urlString) {
+          const result = {
+            success: false,
+            entity: undefined,
+            action: undefined,
+            parameters: {},
+          }
+
+          try {
+            const url = new URL(urlString);
+
+            let urlHash = url.hash;
+            let matches = urlHash.match(/^#\/explorer(?<urlHash>\/.*)?$/);
+
+            if (matches === null) throw new Error(`Invalid URL hash: '${urlHash}'`);
+
+            urlHash = matches.groups.urlHash;
+
+            if (urlHash === undefined) return result;
+
+            matches = urlHash.match(/^\/(?<entity>[a-zA-Z0-9_]+)?(?<urlHash>\/.*)?$/);
+            result.entity = matches.groups.entity;
+            urlHash = matches.groups.urlHash
+
+            if (urlHash === undefined) return result;
+
+            matches = urlHash.match(/^\/(?<action>[a-zA-Z0-9_]+)?(?<urlHash>\?.*)?$/);
+            result.action = matches.groups.action;
+            urlHash = matches.groups.urlHash
+
+            if (urlHash === undefined) return result;
+
+            result.parameters = parseApiCallParams(urlHash);
+            result.success = true;
+          } catch (error) {
+            console.error("Invalid APIv4 Explorer URL:", urlString);
+            console.error(error);
```

Comment:
Can we make this error user-visible? Maybe just a simple `CRM.alert` or something like that. It's currently a bit hard to see if changes are applied when there are already existing values in the action.